packages/core-mobile/app/new/routes/index.tsx

```diff
@@ -1,41 +1,22 @@
-import { useRouter } from 'expo-router'
-import { View, Button } from '@avalabs/k2-alpine'
+import { useTheme } from '@avalabs/k2-alpine'
 import React from 'react'
-import { showNotificationAlert, showSnackbar } from 'new/utils/toast'
+import { View } from 'react-native'
 
-export default function Index(): JSX.Element {
-  const { navigate } = useRouter()
-
-  const handleSignIn = (): void => {
-    navigate('/portfolio/')
-
-    showSnackbar('Welcome back!')
-  }
-
-  const handleSignUp = (): void => {
-    navigate('/signup/')
-
-    showNotificationAlert({
-      type: 'success',
-      title: 'Sign up',
-      message: 'Sign up button pressed'
-    })
-  }
+function IndexScreen(): JSX.Element {
+  const { theme } = useTheme()
 
   return (
     <View
       style={{
-        flex: 1,
+        width: '100%',
+        height: '100%',
+        backgroundColor: theme.colors.$surfacePrimary,
         alignItems: 'center',
         justifyContent: 'center',
-        gap: 16
-      }}>
-      <Button type="primary" size="medium" onPress={handleSignIn}>
-        Sign in
-      </Button>
-      <Button type="primary" size="medium" onPress={handleSignUp}>
-        Sign up
-      </Button>
-    </View>
+        position: 'absolute'
+      }}
+    />
```

Comment:
I’ve set up a blank screen at the top level. The reason is that when navigation occurs due to walletState, there’s a slight delay, causing the top screen to appear very briefly (hardly noticeable on iOS and for less than 500ms on Android). It would be great to eliminate the flicker entirely, but I haven’t found a way to do that yet. In the meantime, showing the blank screen during the flicker felt the most natural.